cpp/CMakePresets.json

```diff
@@ -46,6 +46,32 @@
         "CMAKE_BUILD_TYPE": "RelWithDebInfo"
       }
     },
+    {
+      "name": "emscripten-overrides",
+      "hidden": true,
+      "cacheVariables": {
+        "ARROW_BUILD_SHARED": "OFF",
+        "ARROW_BUILD_STATIC": "ON",
+        "ARROW_BUILD_TESTS": "OFF",
+        "ARROW_ENABLE_THREADING": "OFF",
+        "ARROW_CUDA": "OFF",
+        "ARROW_MIMALLOC": "OFF",
+        "ARROW_JEMALLOC": "OFF",
+        "ARROW_S3": "OFF",
+        "ARROW_DEPENDENCY_SOURCE": "BUNDLED",
+        "ZLIB_SOURCE": "SYSTEM",
+        "ARROW_IPC": "OFF",
```

Comment:
IPC is not only used for network functionality (I see you mentioned below that "``ARROW_IPC`` and anything else that uses network probably won't work."), so this can be enabled by default? Then, for example, reading an Arrow IPC file should work.

Reply:
oh yeah, my mistake, I was thinking of flight.